src/game/engine/taction.cpp

```diff
@@ -0,0 +1,545 @@
+            Map.Set_Tab_Redraw(true);
+            Map.Flag_To_Redraw();
+
+            return true;
+        case TACTION_TIMER_SET:
+            Scen.Set_Global_Time(90 * m_IntegerValue);
+            Map.Set_Tab_Redraw(true);
+            Map.Flag_To_Redraw();
+
+            return true;
+        case TACTION_GLOBAL_SET:
+            Scen.Set_Global_To(m_IntegerValue, true);
+
+            return true;
+        case TACTION_GLOBAL_CLEAR:
+            Scen.Set_Global_To(m_IntegerValue, false);
+
+            return true;
+        case TACTION_AUTO_BASE_AI:
+            hp->Set_Auto_Base_AI(m_IntegerValue != 0);
+
+            return true;
+        case TACTION_GROW_SHROUD:
+            Map.Encroach_Shadow();
+
+            return true;
+        case TACTION_DESTROY_ATTACHED:
+            if (op != nullptr) {
+                int dmg = op->Get_Health();
+                op->Take_Damage(dmg, 0, WARHEAD_AP, nullptr, true);
+            } else {
+                ret = false;
+            }
+
+            if (cell_num != 0) {
+                Map.Destroy_Bridge_At(cell_num);
+            }
+
+            if (tp != nullptr) {
+                for (int i = 0; i < g_Units.Count(); ++i) {
+                    if (&g_Units[i] != nullptr && g_Units[i].Get_Attached_Trigger() == tp) {
+                        g_Units[i].Invalidate_Attached_Trigger();
+                        int dmg = g_Units[i].Get_Health();
+                        g_Units[i].Take_Damage(dmg, 0, WARHEAD_AP, nullptr, true);
+                    }
+                }
+
+                for (int i = 0; i < g_Infantry.Count(); ++i) {
+                    if (&g_Infantry[i] != nullptr && g_Infantry[i].Get_Attached_Trigger() == tp) {
+                        g_Infantry[i].Invalidate_Attached_Trigger();
+                        int dmg = g_Infantry[i].Get_Health();
+                        g_Infantry[i].Take_Damage(dmg, 0, WARHEAD_AP, nullptr, true);
+                    }
+                }
+
+                for (int i = 0; i < g_Aircraft.Count(); ++i) {
+                    if (&g_Aircraft[i] != nullptr && g_Aircraft[i].Get_Attached_Trigger() == tp) {
+                        g_Aircraft[i].Invalidate_Attached_Trigger();
+                        int dmg = g_Aircraft[i].Get_Health();
+                        g_Aircraft[i].Take_Damage(dmg, 0, WARHEAD_AP, nullptr, true);
+                    }
+                }
+
+                for (int i = 0; i < g_Buildings.Count(); ++i) {
+                    if (&g_Buildings[i] != nullptr && g_Buildings[i].Get_Attached_Trigger() == tp) {
+                        g_Buildings[i].Invalidate_Attached_Trigger();
+                        int dmg = g_Buildings[i].Get_Health();
+                        g_Buildings[i].Take_Damage(dmg, 0, WARHEAD_AP, nullptr, true);
+                    }
+                }
+
+                // TODO Again the original code looks bugged related to vessels as they will not be destroyed.
```

Comment:
Its a bug given that it would be unexpected behaviour given how it acts on other unit types. If you were making a map and had a "destroy attached" trigger, you'd expect the attached thing to be destroyed.